### PR TITLE
Add Carousel example. Continue adding CSS abilities.

### PR DESCRIPTION
**Pull request by milljm**
Adjust Carousel to use media directory.
Add working carousel example.
Add ability to use inline CSS.

Modify moose.css:
  Admonitions were running into other elements.
  Adjust `<pre>` CSS, so they do not overflow into floating elements.
  Adjust `<hr>` CSS, so that it actually works.

Add memory_logger output text files instead of having all that listed in Markdown.
Its not only cleaner this way, but allows for additional CSS options.

Refs #7683

### CODE REVIEW

File: python/MooseDocs/extensions/MooseCarousel.py [hunk 1]
```diff
@@ -23,6 +24,24 @@ class MooseCarousel(BlockProcessor, MooseCommonExtension):
...
+      self._settings = {'caption'  : None,
+                        'interval' : None,
+                        'pause'    : None,
+                        'wrap'     : None,
+                        'keyboard' : None}
+
+      # We need a way to limit CSS to elements that we know causes issues.
+      # example: to ignore overflow-y to only div elements
+      # self.invalid_css = { 'div' : ['overflow-y'] }
+      self._invalid_css = {}
```
Comment (aeslaughter) [inline python/MooseDocs/extensions/MooseCarousel.py]: The empty dict() should be provided by the MooseCommonExtension, just like self._settings. If the derived class needs something here, then it populates the dict.

File: python/MooseDocs/extensions/MoosePackageParser.py [hunk 2]
```diff
@@ -27,6 +27,12 @@ def __init__(self, **kwargs):
         self._settings = {'arch' : None,
                           'return' : None}
 
+        # We need a way to limit _some_ CSS that we know causes issues.
+        # example: to ignore overflow-y to only div elements
+        # self.invalid_css = { 'div' : ['overflow-y'] }
+        #
+        self._invalid_css = {}
```
Comment (aeslaughter) [inline python/MooseDocs/extensions/MoosePackageParser.py]: Same as comment above.

File: python/MooseDocs/extensions/MooseTextPatternBase.py [hunk 3]
```diff
@@ -29,26 +29,20 @@ def __init__(self, pattern, language=None, repo=None, root=None, **kwargs):
...
+                          'repo_link'           : True,
+                          'label'               : True,
+                          'method'              : True,
+                          'block'               : True,
+                          'strip-extra-newlines': False}
+
+        # We need a way to limit CSS to elements that we know causes issues.
+        # example: to ignore overflow-y to only div elements
+        # self.invalid_css = { 'div' : ['overflow-y'] }
+        #
```
Comment (aeslaughter) [inline python/MooseDocs/extensions/MooseTextPatternBase.py]: You can remove this comment, this is what should be in the base MooseCommonExtension. The comment below this is good.

File: docs/content/moose_tools/memory_logger.md [hunk 4]
```diff
@@ -11,25 +11,8 @@ memory_logger.py --run "mpiexec -n 2 /absolute/path/to/moose_test-opt -i simple_
...
-12 May 09:24:33.088981        259,356 K | --------------------------------------------------------| 98.14%
-12 May 09:24:33.353685        261,136 K | ---------------------------------------------------------| 98.81%
-12 May 09:24:33.617438        263,812 K | ---------------------------------------------------------| 99.83%
-12 May 09:24:33.881309        264,208 K | ---------------------------------------------------------| 99.98%
-12 May 09:24:34.145606        264,260 K | ------------------------------------------------------------| 100%
-12 May 09:24:34.410900              0 K | | 0%
-Date Stamp                 Memory Usage | Percent of MAX memory used: ( 264,260 K )
-```
+!text docs/content/moose_tools/memory_logger_output1.txt overflow-y=scroll max-height=500px font-size=smaller
+
```
Comment (aeslaughter) [inline docs/content/moose_tools/memory_logger.md]: This is way better!

File: python/MooseDocs/extensions/MooseCarousel.py (hunk 1 above)
Comment (milljm) [inline python/MooseDocs/extensions/MooseCarousel.py]: I thought I was entering that too many times...

**Pull request merged**